pkg/api/v1alpha1/tinkerbelltemplateconfig.go

```diff
@@ -0,0 +1,84 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1/thirdparty/tinkerbell"
+)
+
+const TinkerbellTemplateConfigKind = "TinkerbellTemplateConfig"
+
+func NewTinkerbellTemplateConfigGenerate(name string, opts ...ActionOpt) *TinkerbellTemplateConfigGenerate {
```

Comment:
Why is this limited to action customization's only?

Reply:
Yea I agree with what you are saying.

So I think the idea here was `NewTinkerbellTemplateConfigGenerate` is meant to be used while generating the "default" TemplateConfig as part of the `generate clusterconfig`. In that context we want to keep a single default task and add the default actions. Hence i added options only for actions, since we might want to add/remove those as we develop.

1. With a higher level `NewTinkerbellTemplateConfigGenerateOption` my concern was how do we feed it an action option? Like then we might need a TasksOpt and an ActionOpt to specify what task and what action to add.
2. I could do that. But do you mean like get the config object and then apply defaults?